fast_response/MonitoringAndMocks/Data_Display.py

```diff
@@ -8,756 +8,387 @@
 
-path=os.environ.get('FAST_RESPONSE_OUTPUT')
+path = os.environ.get('FAST_RESPONSE_OUTPUT')
 out_put = '/data/user/jthwaites/o4-mocks/'
 
 #Creating readable (and callable) files from ALL pickle files previously created in gw_gcn_listener
-if pwd.getpwuid(os.getuid())[0] == 'realtime':
-    Pickle_to_text = sorted(glob.glob(path+'PickledMocks/*MS*.pickle'))#+
-                            #glob.glob('/data/user/jthwaites/FastResponseAnalysis/output/PickledMocks/*MS*.pickle'))
-else:
-    Pickle_to_text = sorted(glob.glob(path+'PickledMocks/*MS*.pickle'))
-
-all_dictionary = {'Trigger_Time': [], 'GCN_Alert': [], 'End_Time': [],
-                  'LVK_Latency': [], 'IceCube_Latency': [], 'Total_Latency': [],
-                  'We_had_to_wait:': [], 'Name': [], 'Time_Stamp': []}
-print('Number of mocks to load: {}'.format(len(Pickle_to_text)))
-
-i=0
-for file in Pickle_to_text:
-    #if i%100==0: print(i)
-    if os.path.exists(file):
-        with open(file, 'rb') as pickle_now:
-            Entries = pickle.load(pickle_now)
-        if int(Entries["Trigger_Time"]) == 59957:
-            continue #This is January 13th 2023, dates were messed up.
-        for Key in Entries.keys():
-            if Key=="Ligo_Latency":
-                all_dictionary['LVK_Latency'].append(Entries[Key])
+mock_files = sorted(glob.glob(path+'PickledMocks/*MS*.pickle'))
+
+def sort_mocks(mock_files):
+    event_dict = pd.DataFrame({"Trigger_Time": [], "GCN_Alert": [], "End_Time": [],
+                "LVK_Latency": [], "IceCube_Latency": [], "Total_Latency": [], 
+                "Name": [], "time_stamp": [], "first": [], 
+                "last": [], "we_had_to_wait": []})
+    ###these keys are saved in pickle files and just need to be loaded into dict
+    select_keys = ("Trigger_Time", "GCN_Alert", "End_Time", "LVK_Latency",
+                    "IceCube_Latency", "Total_Latency")
+    ###create empty list and load each value into the list which is then loaded into the dict.
+    loading_dict = {"Trigger_Time": [], "GCN_Alert": [], "End_Time": [],
+                "LVK_Latency": [], "IceCube_Latency": [], "Total_Latency": [], 
+                "Name": [], "time_stamp": [], "first": [], 
+                "last": [], "we_had_to_wait": []}
+    
+    for i in mock_files:
+        mocks = pd.read_pickle(f'{i}')
+        ###load data from pickle files to dataframe
+        for key in select_keys:
+            mock_key = "Ligo_Latency" if key == "LVK_Latency" else key
+            loading_dict[key] = mocks[mock_key]
+
+        ###separate name of event
+        ###file name format: gw_latency_dict_S240501dr-2-Preliminary_test.pickle
+        first_split = i.split('_')
+        second_split = first_split[3].split('-')
+        loading_dict["Name"] = second_split[0]
+
+        ###collect timestamps of creation for mock files
+        timestamp = os.path.getmtime(i)
+        datestamp = datetime.datetime.fromtimestamp(timestamp, datetime.UTC)
```

Comment:
the version of datetime in the virtual env doesn't like `datetime.UTC`. Can you do `timezone.utc` instead?